tools/html_transformer/html_transformer.js

```diff
@@ -0,0 +1,105 @@
+var Promise = require('es6-promise').Promise;
+var jsdom = require('jsdom');
+
+exports.load = function(filename) {
+  return new Promise(function(resolve, reject) {
+    jsdom.env(filename, function(errors, window) {
+      if (errors && errors.length) {
+        return reject(errors[0]);
+      }
+
+      return resolve(new HTMLTransformer(window));
+    });
+  });
+};
+
+function HTMLTransformer(window) {
+  this.window = window;
+}
+exports.HTMLTransformer = HTMLTransformer;
```

Comment:
And for me it would be nice to collect all exports at the bottom of file, not scattering like this.